imucapture/src/main.cpp

```diff
@@ -40,29 +41,116 @@
 #include <iomanip>
 #include <stdexcept>
 #include <string>
-#include <fstream>  //std::ofstream
+#include <fstream>
 #include <ctime>
 
 #include <cstdio>
+#include <stdlib.h>
 #include <termios.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/time.h>
 
-void changemode(int dir){
+void changemode(int dir);
+int kbhit();
+
+void device_initialization(DeviceClass &device, XsPortInfo &mtPort);
+XsPortInfo scan_usb_devices();
+
+void set_message_to_packet(
+	XsDataPacket &packet,
+	XsMessage &msg,
+	XsDeviceId dev_id);
+
+inline void
+print_position_to_file(std::ofstream &outfile, XsDataPacket &packet);
```

Comment:
Could these functions be unified under an operator<<?

Reply:
That's possible. XsDataPacket is just a struct, so all that would need to be added is the operator overload function.